scapy/layers/bluetooth4LE.py

```diff
@@ -199,14 +199,35 @@ class BTLE_DATA(Packet):
         BitField("SN", 0, 1),
         BitField("NESN", 0, 1),
         BitEnumField("LLID", 0, 2, {1: "continue", 2: "start", 3: "control"}),
-        ByteField("len", None),
+        ByteField("len", None), # BLE 4.2 and upwards can use 1 entire byte for length
     ]
 
     def post_build(self, p, pay):
         if self.len is None:
             p = p[:-1] + chb(len(pay))
         return p + pay
 
+    def do_dissect_payload(self, s):
+        if s is not None:
+            cls = self.guess_payload_class(s)
+            try:
+                p = cls(s, _internal=1, _underlayer=self)
+            except KeyboardInterrupt:
+                raise
+            except Exception:
+                if conf.debug_dissector:
+                    if issubtype(cls, Packet):
+                        log_runtime.error("%s dissector failed" % cls.__name__)
+                    else:
+                        log_runtime.error("%s.guess_payload_class() returned [%s]" % (self.__class__.__name__, repr(cls)))  # noqa: E501
+                    if cls is not None:
+                        raise
+                p = conf.raw_layer(s, _internal=1, _underlayer=self)
+            self.add_payload(p)
```

Comment:
This seems to be a duplicate the default func.

Reply:
Yes, gonna change this by using dissection_done as proposed by @guedou